snuba/query/conditions.py

```diff
@@ -112,3 +116,11 @@ def is_unary_condition(exp: Expression, operator: str) -> bool:
         and exp.function_name == operator
         and len(exp.parameters) == 1
     )
+
+
+def is_condition(exp: Expression) -> bool:
+    return (
+        isinstance(exp, FunctionCall)
+        and exp.function_name in FUNCTION_TO_OPERATOR
+        and len(exp.parameters) > 0
+    )
```

Comment:
Here as well you can use the matcher and return "relatively" strictly typed parameters avoiding further processing on the expression.